src/test/scala/tech/cryptonomic/conseil/tezos/PlatformDiscoveryOperationsTest.scala

```diff
@@ -163,4 +171,83 @@ class PlatformDiscoveryOperationsTest
     }
   }
 
+  "listAttributeValues" should {
+
+    "return list of values of kind attribute of Fees without filter" in {
+      val avgFee = AverageFees(1, 3, 5, Timestamp.valueOf(LocalDateTime.of(2018, 11, 22, 12, 30)), "example1")
+
+      Await.result(dbHandler.run(TezosDatabaseOperations.writeFees(List(avgFee))), Duration.Inf)
+      dbHandler.run(
+        DBIO.sequence(
+          PlatformDiscoveryOperations.verifyAttributesAndGetQueries("fees", "kind", None)
+        )
+      ).futureValue.flatten shouldBe List("example1")
+    }
+
+    "return throw an exeption when asked for medium attribute of Fees without filter - numeric attributes should not be displayed" in {
+      val avgFee = AverageFees(1, 3, 5, Timestamp.valueOf(LocalDateTime.of(2018, 11, 22, 12, 30)), "example1")
+
+      Await.result(dbHandler.run(TezosDatabaseOperations.writeFees(List(avgFee))), Duration.Inf)
+
+      val thrown = intercept[Exception] {
+        dbHandler.run(
+          DBIO.sequence(
+            PlatformDiscoveryOperations.verifyAttributesAndGetQueries("fees", "medium", None)
+          )
+        ).futureValue
+      }
+      thrown.getMessage shouldBe "The future returned an exception of type: java.util.NoSuchElementException, with message: Action.withFilter failed."
```

Comment:
I guess there should be a way to extract a failed future's Exception type and check that without depending on a specific string message

Reply:
I can only catch `org.scalatest.exceptions.TestFailedException` from what I see which hides my expected exception, so I can check for this exception but I would still need to check for the message - I can change to checking for `TestFailedException`- WDYT?

Reply:
You can always call `future.failed.futureValue` to get back the exception that failed the future.

Then you check the type with `should be a[NoSuchElementException]`

At least... I guess it should work

Reply:
with `future.failed.futureValue` it does not throw any exception:
`Expected exception java.util.NoSuchElementException to be thrown, but no exception was thrown`.

Reply:
ok, I found `future.value.get` which returns exception it supposed to

Reply:
`val ex = future.failed.futureValue` is supposed to be the exception itself, not to throw the exception... 😀